api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -456,6 +456,16 @@ type HostedClusterSpec struct {
 	// +required
 	Platform PlatformSpec `json:"platform"`
 
+	// KubeAPIServerDNSName specifies a desired DNS name to resolve to the KAS.
+	// This can only be set if secretRef is also set. When set, the controller will automatically generate a secret with kubeconfig and expose it in the hostedCluster Status.
+	// If it's set or removed day 2, all generated objects will be created, recreated or deleted.
```

Comment:
can we be specific i.e. r/"all generated objects"/ "the kubeconfig generated secret"?